### PR TITLE
Update Searcher and Symbols to use StatefulSets

### PR DESCRIPTION
**Pull request by scjohns**
### Checklist

- [ ] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [ ] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)

### Test plan

1. `helm lint charts/sourcegraph/.`
2. `helm template charts/sourcegraph`
3. Installed in minikube with no errors

### CODE REVIEW

**Review by loujar — APPROVED** [review]
ServiceNow has successfully deployed their non-prod sourcegraph instance with these modifications, validating the changes.

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml [hunk 1]
```diff
@@ -116,9 +108,21 @@ spec:
...
       {{- end }}
+  volumeClaimTemplates:
+  - metadata:
+      name: cache
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          # The size of disk used to mirror your git repositories.
```
Comment (abeatrix) [inline charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:125]:
```suggestion

```

Remove comment from gitserver

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml [hunk 2]
```diff
@@ -116,9 +108,21 @@ spec:
...
+          # If you change this, also change indexed-search's disk size.
```
Comment (abeatrix) [inline charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:126]:
```suggestion

```

Remove comment from gitserver, not applicable for searcher / symbols

File: charts/sourcegraph/templates/symbols/symbols.StatefulSet.yaml [hunk 3]
```diff
@@ -124,9 +116,21 @@ spec:
...
       {{- end }}
+  volumeClaimTemplates:
+  - metadata:
+      name: cache
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          # The size of disk used to mirror your git repositories.
```
Comment (abeatrix) [inline charts/sourcegraph/templates/symbols/symbols.StatefulSet.yaml:133]:
```suggestion

```
Remove comment from gitserver --not applicable for searcher / symbols.

File: charts/sourcegraph/templates/symbols/symbols.StatefulSet.yaml [hunk 4]
```diff
@@ -124,9 +116,21 @@ spec:
...
+          # If you change this, also change indexed-search's disk size.
```
Comment (abeatrix) [inline charts/sourcegraph/templates/symbols/symbols.StatefulSet.yaml:134]:
```suggestion

```

Remove comment from gitserver --not applicable for searcher / symbols.

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml [hunk 5]
```diff
@@ -89,7 +81,7 @@ spec:
           timeoutSeconds: 5
         volumeMounts:
         - mountPath: /mnt/cache
-          name: cache-ssd
+          name: cache
```
Comment (abeatrix) [inline charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:84]: Should we keep cache-ssd as the volume name? We are also using cache-ssd in the kustomize deployment. What do you think?

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml [hunk 6]
```diff
@@ -89,7 +81,7 @@ spec:
...
         - mountPath: /tmp
```
Comment (abeatrix) [inline charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:85]: What's this tmp dir for?

**Review by abeatrix — CHANGES_REQUESTED** [review]
Saw some comments that might be copied and pasted from gitserver to the new symbols and searcher by mistakes that need to be removed, but overall lgtm!

I do have a question about the new tmp dir though. I don't think symbols and searcher need one since all the data are stored in cache-ssd?

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml (hunk 6 above)
Comment (scjohns) [reply charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:85]: Not sure what it's used for, but it was in the deployment originally, so I kept it

Comment (scjohns) [reply charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:85]: It was added [here](https://github.com/sourcegraph/deploy-sourcegraph-helm/commit/772bf4739649df7323a358e201416ef685f74098) it looks like a while back. If we know it isn't used, I'm fine to delete, but wanted to keep consistent from what was created in the past.

File: charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml [hunk 7]
```diff
@@ -59,11 +57,7 @@ spec:
           {{- $item | toYaml | nindent 10 }}
         {{- end }}
         - name: SEARCHER_CACHE_SIZE_MB
-          valueFrom:
-            resourceFieldRef:
-              containerName: searcher
-              resource: requests.ephemeral-storage
-              divisor: 1M
+          value: {{ trimSuffix "G" .Values.searcher.storageSize | mul 500 | quote }}
```
Comment (scjohns) [inline charts/sourcegraph/templates/searcher/searcher.StatefulSet.yaml:60]: @abeatrix accounted for the cache size env variable now.

```
❯ k exec -it symbols-0 -- /bin/sh
/ $ env| grep "SYMBOLS_CACHE_SIZE_MB"
SYMBOLS_CACHE_SIZE_MB=6000
/ $ exit
```

set properly in the pod

**Review by abeatrix — APPROVED** [review]

**Comment by scjohns** [comment]
@abeatrix @loujar thoughts on [698633c](https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/242/commits/698633c315eff2ddea694b80bda7fa541c012550)?

**Comment by scjohns** [comment]
Validated latest changes with full k8s install, search worked with deployments on default helm chart with 4.3.0 and with changes after upgrade to 4.4.2 (+ statefulSets)

**Comment by daxmc99** [comment]
@loujar What version of Kubernetes is service now running with?

**Comment by loujar** [comment]
@daxmc99 ServiceNow applied these changes in their v4.3.1 Sourcegraph deployment

**Pull request merged**